blog/2018/12/07/openapi/index.markdown

```diff
@@ -0,0 +1,169 @@
+# MetaCPAN, Mojolicious and OpenAPI
+
+During this years [meta::hack 3](http://www.olafalders.com/2018/11/21/metahack-3-wrap-report/), I was extremely fortunate to work with Joel Berger on integrating/documenting OpenAPI with the MetaCPAN API via Mojolicious.
+
+## What is it?
+OpenAPI is a specification for designing, documenting, validating and driving your RESTful API. It can be used to provide documentation to an existing API, or staring fresh.
+
+The OpenAPI Specification originated as the Swagger specification and was renamed to separate the API description format from the open source tooling. The specification moved into new GitHub repository, but did not change.
+
+In the case of the MetaCPAN API, we set out to provide documentation to the existing API, but quickly moved into supporting validation to API calls as well.
+
+---
+
+## The tools
+There are many tools available for working with OpenAPI, including discovery tools that will assist in writing your OpenAPI specification. Our choice was to write the definition by hand (in vim of course) and use tools to generate the documentation, and integrate the specification into MetaCPAN.
+
+* [ReDoc](https://github.com/Rebilly/ReDoc) — OpenAPI/Swagger-generated API Reference Documentation
+
+We chose ReDoc to provide the human documentation interface. ReDoc provides an [HTML template](https://github.com/Rebilly/ReDoc#tldr)) to be served as a static file. The template creates an interactive page providing documentation and examples based on the details provided in the OpenAPI specification file.
+
+* [Mojolicious::Plugin::OpenAPI](https://metacpan.org/pod/Mojolicious::Plugin::OpenAPI) — OpenAPI / Swagger plugin for Mojolicious
+
+Reads the OpenAPI specification file and adds the appropriate routes and validations for your Mojolicious based application.
+
+* [JSON::Validator](https://metacpan.org/pod/JSON::Validator) — Validate data against a JSON schema
+
+Integrated into the Mojolicious::Plugin::OpenAPI module, provides the input and output validation, as well as providing validation for the specification file itself.
+
+## Getting Started
+The following strategy was used when implementing the MetaCPAN OpenAPI specification. 
+
+### The OpenAPI Specification File
+
+The specification can either be written in JSON or YAML. With support for multiline attribute values making it much easier to read and write with less formatting, we used YAML. 
```

Comment:
You're burying the lede here. "The specification can be written in [...]. We chose YAML because ... "